### ISSUE TITLE
Repository structure and contents

### ISSUE THREAD

**Issue by masbicudo** [gearz-lab/react-ui#2]
What is all that stuff in the repository?

**HTML5 boiler plate**

I don't think that _HTML5 boiler plate_ should be in the root of this project.
I don't think it should be in this project in the first place.

If we want to make a site with live samples, we could make a
[github page](https://pages.github.com), and commit _HTML5 boiler plate_ there instead.

Also when creating an issue this appears at top, caused by the existence of `CONTRIBUTING.md`:

![image](https://cloud.githubusercontent.com/assets/1653710/6090315/9454edac-ae5b-11e4-9cfd-fb8d4c21bfb7.png)

**.idea**

What are these files?

They seem to be some kind of cached information from a tool that you might be using.
Please, don't commit them if you don't think these files should be shared with others.

In some projects I commit Resharper `.DotSettings` files and `Settings.StyleCop`.
They are coding style settings that I want to share with other developers.
If these `.idea` files have this purpose then I think it is ok, but by looking at their contents, it doesn't appear to be that.
## Structure

Here is a structure that fulfils the purpose of this repository:
- `components\` - only the code of our components: scripts and styles made by ourselves
  - `default\` - the default presentation structure (our own)
    - `pager.jsx` - the pager component rendered with our own structure
    - `skins\` - skins for our presentational structure
      - `light\` - light themed skin
        - `pager.css` - skin for the pager component
      - `dark\` - dark themed skin
        - `pager.css` - skin for the pager component
  - `bootstrap-3\` - components using the bootstrap-3 presentation framework
    - `pager.jsx` - pager component rendered for bootstrap-3
    - `skins\` - shouldn't exist, since bootstrap has it's own skins
- `samples\` - samples of component usage
  - `pager-basic-samples\`
    - `pager-default.html` - sample using the pager
    - `pager-bootstrap-3.html` - sample using the pager with bootstrap-3 structure...
      the code is the same as the above file, but including the corresponding pager.jsx file
    - `pager-sample.js` - script compatible with both structures (because they can be swapped)

**Notes:**
- Components in different folders, but with the same name, must have the same function,
  and they must be swappable:
  - `components\default\pager.jsx` and `components\bootstrap-3\pager.jsx` behave exactly the same,
    only the HTML structure is different between them.
  - names inside JavaScript should also be the same
- The samples should be built with component swapping in mind.
- When components are not swappable, then they must have different names, both file and inside
  the JavaScript.

Thoughts?

**Comment by andrerpena** [comment]
I agree. Except that I would put the components inside a `src` folder because it seems like there's a tool called `gulp` that is widely used and it allows you to _compile_ your code into a `dist` folder.

**Comment by andrerpena** [comment]
I'm including the .idea folder in the .gitignore

**Comment by masbicudo** [comment]
Nice, I totally agree.
